app/controllers/recipes_controller.rb

```diff
@@ -22,6 +27,6 @@ def show
   private
 
   def recipe_params
-    params.require(:recipe).permit(:name, :style, :yeast, :summary, :notes)
+    params.require(:recipe).permit(:name, :style, :grains, :yeast, :summary, :notes)
```

Comment:
yep, that did it. I had to permit `grains_attributes: [:name, :weight]`. Thanks!